libs/common/src/main/kotlin/person/Person.kt

```diff
@@ -128,9 +130,19 @@ data class VergeEllerFullmektig(
     val omfangetErInnenPersonligOmraade: Boolean
 )
 
-data class FolkeregisterIdent(
-    val folkeregisterident: Folkeregisteridentifikator
-)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+sealed class PdlIdentifikator {
```

Comment:
For meg ser det ut som at alt dette brukes bare av denne ene tjenesten. Gir dette så mye sammenlignet med å bare ha noe enkelt som for eksempel `data class PDLident(val type: String, val ident: String)`

Reply:
Jeg vil si det, siden for å håndtere det enkle tilfellet legger du en større byrde på alle som bruker klassen til å forstå hva de kan få / putte inn, og dermed må ha logikk for det uansett. Med en sealed class får du ganske klar info om at denne er enten det ene eller det andre, og garanterer at det er kun de tilfellene du må håndtere

Reply:
Eneste logikken som blir borte om dette fjernes er valideringen av at identen oppfyller reglene for den type identer pdl sier det er? I utgangspunktet tenker jeg at det er PDL sitt ansvar. De skal ikke sende oss fnr som ikke oppfyller reglene for fnr. Garantien du nevner er kjekk når definisjon og bruk er innenfor samme "kompileringsenhet", men på ting som serialiseres og deserialiseres er jeg mer skeptisk. EN konsekvens er jo at utvidelser ikke er bakoverkompatible. Ser ikke det store problemet med å ha det slikt her da.